warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -5484,11 +5493,17 @@ public static void reset(GeoTimeSerie gts) {
 
         for (int j = 0; j < subgts.values; j++) {
           ((Object[]) parms[6])[j] = valueAtIndex(subgts, j);
-          if (-1 == tickidx && tick == tickAtIndex(subgts, j)) {
+          if ((-1 == tickidx || reversed) && tick == tickAtIndex(subgts, j)) {
```

Comment:
Maybe add a comment why reversed is checked here, from the code it is not obvious.